src/lib/input/input.ts

```diff
@@ -351,7 +350,26 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
    * Implemented as part of MatFormFieldControl.
    * @docs-private
    */
-  get shouldLabelFloat(): boolean { return this.focused || !this.empty; }
+  get shouldLabelFloat(): boolean {
+    if (this._elementRef.nativeElement.nodeName.toLowerCase() === 'select') {
+      return this.shouldLabelFloatForSelect(this._elementRef.nativeElement);
+    } else {
+      return this.focused || !this.empty;
+    }
+  }
+
+  private shouldLabelFloatForSelect(selectElement:HTMLSelectElement): boolean {
+    if (selectElement.multiple) {
+      // For multi select, float mat input label
+      return true;
+    } else if (!this.empty || selectElement.options[0].label || selectElement.options[0].innerHTML) {
```

Comment:
Since this is being called in a getter, it can be very expensive to keep checking `innerHTML`. We should either move away from checking the DOM or check `textContent`.

Reply:
we can just use label to cover both.  from the documentation "label
This attribute is text for the label indicating the meaning of the option. If the label attribute isn't defined, its value is that of the element text content."
And testing confirmed that.

Reply:
I think this method can be simplified into one statement:
```ts
return selectedElement.multiple || this.focused || !this.empty ||
    selectElement.options[0].label || selectElement.options[0].innerHTML;
```